Resources/input/fsh/profiles/3.1_Prescription_ServiceRequest.fsh

```diff
@@ -0,0 +1,111 @@
+
+* status MS
+* status obeys sr-1
+* status from GEM_VS_MEDREQ_RequestStatus
+  * ^short = "Gibt den Bearbeitungsstand eines ServiceRequests an"
+
+* intent = #order (exactly)
+
+* code MS
+  * ^short = "Gibt die Art des ServiceRequests an"
+  * ^comment = "#prescription-request dient der Anfrage eines Rezeptes an einen Arzt"
+* code.coding 1..1 MS
+* code.coding.system 1..1
+* code.coding.system = "https://gematik.de/fhir/erpmedreqcom/CodeSystem/GEM-CS-MEDREQ-ServiceRequest-Code" (exactly)
+* code.coding.code 1..1
+* code.coding.code = #prescription-request (exactly)
+
+* subject MS
+* subject only Reference($KBV_PR_FOR_Patient)
+  * ^short = "Patient für den stellvertretend ein Rezept angefordert wird"
+
+* orderDetail 1..1 MS
+* orderDetail.coding 1..1 MS
+* orderDetail.coding from GEM_VS_MEDREQ_Prescription_Fullfillment_Type
+  * ^short = "Gibt an wie die Rezeptanfrage erfüllt werden soll."
+  * ^comment = "Wenn das Rezept an eine Apotheke geht, muss ein entsprechender ServiceRequest im Bundle enthalten sein."
+
+* occurrence[x] 1..1 MS
+* occurrence[x] only dateTime
+  * ^short = "Gibt das Datum an, an dem das Rezept ausgestellt werden soll"
+
+* authoredOn 1..1 MS
+  * ^short = "Erstellungsdatum der Anfrage"
+  * ^comment = "Wird initial angelegt und dann nicht mehr verändert."
+
+
+* requester 1..1 MS
+* requester only Reference(GEM_PR_MEDREQ_Organization or $KBV_PR_FOR_Practitioner)
+  * ^short = "Anfragende Einrichtung oder Arzt"
+
+* performer MS
+* performer only Reference($KBV_PR_FOR_Practitioner)
+  * ^short = "Arzt, der das Rezept ausstellen soll"
+
+* reasonCode MS
+* reasonCode from GEM_VS_MEDREQ_MedicationRequest_Reason
+  * ^short = "Code, der angibt, warum das Medikament angefragt wird"
+
+* reasonReference only Reference(GEM_PR_ERP_MEDREQ_RemainingMedication)
+  * ^comment = "Wenn eine Referenz zum Grund angegeben wird muss auch ein menschen lesbarer Eintrag in .note hinzugefügt werden, falls Systeme die referenzierte Observation nicht verarbeiten können"
+
+* supportingInfo ^slicing.discriminator.type = #pattern
+* supportingInfo ^slicing.discriminator.path = "type"
+* supportingInfo ^slicing.rules = #open
+* supportingInfo ^slicing.description = "Unterstützende Informationen zur Rezeptanforderung"
+
+* supportingInfo MS
+* supportingInfo contains
+AuslieferndeApotheke 0..1 MS
+* supportingInfo[AuslieferndeApotheke] only Reference(GEM_PR_MEDREQ_Organization)
+* supportingInfo[AuslieferndeApotheke].type = "Organization"
+
+* note MS
+  * ^short = "Weitere Angaben zur Rezeptanforderung"
+  * ^comment = "Eventuell nicht spezifizierte Anwendungsfälle können hier im Freitext platziert werden"
+
+Invariant: sr-1
+Description: "Wenn der Status auf 'completed' gesetzt ist, muss ein Token vorhanden sein"
+Expression: "(status = 'completed') implies extension('https://gematik.de/fhir/erpmedreqcom/StructureDefinition/GEM-EX-MEDREQ-EPrescriptionToken').exists()"
```

Comment:
Diese invariante funktioniert wie gewünscht auf fhirpath playground. 
Aber Hapi meckert da, obwohl es valide sein sollte

Reply:
Vermutlich ist die FhirPath Java Version der Pipeline nicht auf dem neusten Stand. extension() wurde früher in java nicht supported. Workaround:
extension.where(url = '[url]')

Reply:
Ich habe keine Push Rechte, darum hier die korrigierte Version:
`Expression: "(status = 'completed') implies extension.where(url = 'https://gematik.de/fhir/erpmedreqcom/StructureDefinition/GEM-EX-MEDREQ-EPrescriptionToken').exists()"`

Reply:
Hallo @patrick-werner, 
danke für deine Mitarbeit und Unterstützung! 
Auch das funktioniert bei mir mit dem HAPI leider nicht. Wenn Hapi läuft, dann meldet er bei einem Prescription-ServiceRequest, der den Status "active" hat einen Fehler: 

"Bundle.entry[1].resource.ofType(ServiceRequest).status | Rule sr-1: 'Wenn der Status auf 'completed' gesetzt ist, muss ein Token vorhanden sein' Failed"

Funktioniert es wenn du das mit HAPI validierst? Konkretes Beispiel wäre `Bundle-UC1-1-Prescription-Request-To-Prescriber.json`

Reply:
Ich werde das Problem heute in hapi nachvollziehen und korrigieren.

Reply:
Hallo @patrick-werner,
Habe dieses Anliegen in ein Issue verwandelt: https://github.com/gematik/spec-erezept-medicationrequest-communication/issues/24
Konntest du schon was Neues herausfinden?